From 85eb38c356f7fe222362f48f775011548225b22e Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Tue, 30 Apr 2013 14:06:14 +0800
Subject: [PORT FROM MAIN] Some mkv clips can't be recognized in Gallery

BZ: 48944

The Matroska Extractor fails to recognize the
"V_MS/VFW/FOURCC" Codec ID. The patch adds support
for this CodecID.

Original patch: R3 30130

Change-Id: I044f9de803cb9b1b00c064186e30bca1eb3c8496
Orig-Change-Id: I2d567dac5c10068607d89211d1cfc307ff95d545
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: wfeng6 <wei.feng@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../libstagefright/matroska/MatroskaExtractor.cpp  |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/matroska/MatroskaExtractor.cpp b/media/libstagefright/matroska/MatroskaExtractor.cpp
index b304749..6c95706 100644
--- a/media/libstagefright/matroska/MatroskaExtractor.cpp
+++ b/media/libstagefright/matroska/MatroskaExtractor.cpp
@@ -871,7 +871,10 @@ void MatroskaExtractor::addTracks() {
                     }
                 } else if (!strcmp("V_VP8", codecID)) {
                     meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_VPX);
-                } else {
+                } else if (!strcmp("V_MS/VFW/FOURCC", codecID)) {
+                    meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_H263);
+                }
+                else {
                     ALOGW("%s is not supported.", codecID);
                     continue;
                 }
-- 
1.7.4.1

